src/equations/shallow_water_wet_dry_1d.jl

```diff
@@ -0,0 +1,677 @@
+    H0::RealT      # constant "lake-at-rest" total water height
+    # `threshold_limiter` used in `PositivityPreservingLimiterShallowWater` on water height,
+    # as a (small) shift on the initial condition and cutoff before the next time step.
+    # Default is 500*eps() which in double precision is ≈1e-13.
+    threshold_limiter::RealT
+    # `threshold_wet` applied on water height to define when the flow is "wet"
+    # before calculating the numerical flux.
+    # Default is 5*eps() which in double precision is ≈1e-15.
+    threshold_wet::RealT
+end
+
+# Allow for flexibility to set the gravitational constant within an elixir depending on the
+# application where `gravity_constant=1.0` or `gravity_constant=9.81` are common values.
+# The reference total water height H0 defaults to 0.0 but is used for the "lake-at-rest"
+# well-balancedness test cases.
+# Strict default values for thresholds that performed well in many numerical experiments
+function ShallowWaterEquationsWetDry1D(; gravity_constant, H0 = zero(gravity_constant),
+                                       threshold_limiter = nothing,
+                                       threshold_wet = nothing)
+    T = promote_type(typeof(gravity_constant), typeof(H0))
+    if threshold_limiter === nothing
+        threshold_limiter = 500 * eps(T)
+    end
+    if threshold_wet === nothing
+        threshold_wet = 5 * eps(T)
+    end
+    ShallowWaterEquationsWetDry1D(gravity_constant, H0, threshold_limiter,
+                                  threshold_wet)
+end
+
+Trixi.have_nonconservative_terms(::ShallowWaterEquationsWetDry1D) = True()
+function Trixi.varnames(::typeof(Trixi.cons2cons), ::ShallowWaterEquationsWetDry1D)
+    ("h", "h_v", "b")
+end
+# Note, we use the total water height, H = h + b, as the first primitive variable for easier
+# visualization and setting initial conditions
+function Trixi.varnames(::typeof(Trixi.cons2prim), ::ShallowWaterEquationsWetDry1D)
+    ("H", "v", "b")
+end
+
+# Set initial conditions at physical location `x` for time `t`
+"""
+    initial_condition_convergence_test(x, t, equations::ShallowWaterEquationsWetDry1D)
+
+A smooth initial condition used for convergence tests in combination with
+[`source_terms_convergence_test`](@ref)
+(and [`BoundaryConditionDirichlet(initial_condition_convergence_test)`](@ref) in non-periodic domains).
+"""
+
+function Trixi.initial_condition_convergence_test(x, t,
+                                                  equations::ShallowWaterEquationsWetDry1D)
+    return Trixi.initial_condition_convergence_test(x, t,
+                                                    Trixi.ShallowWaterEquations1D(equations.gravity,
+                                                                                  equations.H0,
+                                                                                  eps(),
+                                                                                  eps()))
```

Comment:
Another kind of "note to self". This constructor to the `Trixi.ShallowWaterEquations1D` will change once the thresholds move

Reply:
Yes, I will add a comment to the file as a reminder.

Reply:
Done in 64b9bf37466ed8503b3f15f4ca63a2f28bbcd9b0